samples/Test.AppWriter/Program.cs

```diff
@@ -20,99 +27,180 @@ private static IServiceProvider ConfigureServiceProvider()
     }
 
-    private static void Main(string[] args)
+    // Attempt to do specified app creation
+    private static void CreateMSApp(string fullPathToMsApp, int numScreens, string[] controlsinfo)
     {
         // Setup services for creating MSApp representation
         var provider = ConfigureServiceProvider();
 
-        var fullPathToMsApp = args.Length > 0 ? args[0] : null;
-        if (fullPathToMsApp == null)
-        {
-            Console.WriteLine("No args provided, using default file path");
-            fullPathToMsApp = Path.Combine(Directory.GetCurrentDirectory(), "CanvasApp.msapp");
-            Console.WriteLine(fullPathToMsApp);
-        }
-
-        if (File.Exists(fullPathToMsApp)) // Overwrite
-        {
-            Console.WriteLine("Warning: File already exists;  Overwrite? (y / n)");
-            var input = Console.ReadLine();
-            if (input?.ToLower()[0] == 'y') File.Delete(fullPathToMsApp);
-        }
-
         // Create a new empty MSApp
         using var msapp = provider.GetRequiredService<IMsappArchiveFactory>().Create(fullPathToMsApp);
 
-        // Add a basic example app (note: this will be replaced with interactive process)
-        msapp.App = GetExampleApp(provider, Path.GetFileNameWithoutExtension(fullPathToMsApp));
+        msapp.App = ExampleAppGenerator.CreateApp(provider, Path.GetFileNameWithoutExtension(fullPathToMsApp), numScreens, ExampleAppGenerator.ParseControlsInfo(controlsinfo));
 
         // Output the MSApp to the path provided
         msapp.Save();
+        Console.WriteLine("Success!  MSApp generated and saved to the provided path");
+    }
+
+    //private static App InteractiveGenerator()
+    //{
+        // do interactive session
+        //var inProgress = true;
+        //while (inProgress)
+        //{
+        //    Console.Write("Create new Screen? (y/n): ");
+        //    var input = Console.ReadLine();
+        //    if (input?.ToLower()[0] == 'y') { } // do custom controls creation
+        //    else
+        //    {
+        //        Console.Write("Create new Screen? (y/n): ");
+        //    }
+        //}
+    //}
+
+    private static void Main(string[] args)
+    {
+        //var = new Option<>(
+        //    name: "",
+        //    description: ""
+        //    );
+        //var interactiveOption = new Option<bool>(
+        //    name: "--interactive",
+        //    description: "Enables interactive mode for MSApp creation"
+        //    );
+        var filePathOption = new Option<FileInfo?>(
+            name: "--filepath",
+            description: "",
+            isDefault: true,
+            parseArgument: result =>
+            {
+                if (result.Tokens.Count == 0)
+                {
+                    result.ErrorMessage = "Must provide a file path.";
+                    return null;
+                    //Console.WriteLine("Using default out path: ");
+                    //return new FileInfo("appname.msapp");
+                }
```

Comment:
If you set the `IsDefault` property on the `Option<>`, it should handle this for you.